client/ayon_core/hosts/zbrush/hooks/create_zmenu_script.py

```diff
@@ -0,0 +1,88 @@
+"""Pre-launch to force zbrush startup script."""
+import os
+import subprocess
+from ayon_core.hosts.zbrush import ZBRUSH_HOST_DIR
+from ayon_core.lib import get_ayon_launcher_args
+from ayon_core.lib.applications import PreLaunchHook, LaunchTypes
+
+
+
+class CreateZMenuScript(PreLaunchHook):
+    """Create AYON Menu Zscript to Zbrush.
+
+    Note that this works in combination whit Zbrush startup script
+    to successfully install zscripts.menu
+
+    Hook `GlobalHostDataHook` must be executed before this hook.
+    """
+    app_groups = {"zbrush"}
+    order = 12
+    launch_types = {LaunchTypes.local}
+
+    def execute(self):
+
+        zscript_path = os.path.join(ZBRUSH_HOST_DIR, "api", "zscripts")
+        os.makedirs(zscript_path, exist_ok=True)
+        zscript_txt = os.path.join(zscript_path, "ayon_zbrush_menu.txt")
+        with open(zscript_txt, "w") as zscript:
+            zscript.write(self.ayon_menu())
+            zscript.close()
+        self.launch_context.env["AYON_ZBRUSH_CMD"] = os.environ["AYON_ZBRUSH_CMD"]
+    def ayon_menu(self):
+        zclient_path = os.path.join(ZBRUSH_HOST_DIR, "api", "widgets.py")
+        zclient_path = zclient_path.replace("\\", "/")
+        os.environ["AYON_ZBRUSH_CMD"] = subprocess.list2cmdline(
+            get_ayon_launcher_args()
+        )
+        python_exe = os.environ["AYON_ZBRUSH_CMD"]
+        ayon_script = ("""
+[ISubPalette,"Zplugin:AYON"]
+[VarSet, addon, "addon"]
+[VarSet, zbrush, "zbrush"]
+[VarSet, zscript, "run-with-zscript"]
+[VarSet, arg, "--launcher"]
+[IButton,"Zplugin:AYON:Load","Loader",
+	[VarSet, loader, "loader_tool"]
+    [VarSet, q, [StrFromAsc, 34]]
+	[VarSet, cmd, [StrMerge, start, " ",#q, #q, "  ",#q, "{py}", #q]]
+    [VarSet, cmd, [StrMerge, cmd, " ", #addon, #q]]
+    [VarSet, cmd, [StrMerge, cmd, #q,  " ", #zbrush, #q]]
+    [VarSet, cmd, [StrMerge, cmd, #q,  " ", #zscript, #q]]
+    [VarSet, cmd, [StrMerge, cmd, #q,  " ", #arg, #q]]
+    [VarSet, cmd, [StrMerge, cmd, #q,  " ", #loader, #q]]
```

Comment:
No idea what this really does but I wonder if this is just more readable if we do not do all this string merging in Zscript per variable but instead just do e.g.

```
[VarSet, cmd, "{py} addon zbrush run-with-script launcher loader_tool"]
[ShellExecute, cmd]
```

Or even:
```
[ShellExecute, "{py}  loader_tool"]
```

Or if we format with e.g.:
```
launch = f"{python_exe} addon zbrush run-with-script launcher "
```

We can just do:
```python
[ShellExecute, "{launch} loader_tool"]
```
If at the end we format the whole string using `launch` variable.
Again, I know nothing of Zscript so there might be reasons not to do this.
Just felt it was greatly unreadable.

Reply:
Would make it something like this:

```python
        launch = f"{python_exe} addon zbrush run-with-script launcher "
        ayon_script = ("""
[ISubPalette,"Zplugin:AYON"]
[IButton,"Zplugin:AYON:Load","Loader",
	[ShellExecute, "{launch} loader_tool"], 0, 120
]
[IButton, "Zplugin:AYON:Publish", "Publish Tab for Publisher",
	[ShellExecute, "{launch} publish_tool"], 0, 120
]
[IButton, "Zplugin:AYON:Manage", "Scene Inventory Manager",
	[ShellExecute, "{launch} scene_inventory_tool"], 0, 120
]
[IButton, "Zplugin:AYON:Workfile", "Workfile",
	[ShellExecute, "{launch} workfiles_tool"], 0, 120
]""").format(py=python_exe)
```

No idea whether that's valid Zscript though 🗡️

Reply:
we can try but i agree it is ugly to read.

Reply:
Looking at the docs you should be able to pass the full command line string to `ShellExecute` directly - since we're building this from Python I'd just to that actually so that this here becomes way more readable.

Reply:
> Looking at the docs you should be able to pass the full command line string to `ShellExecute` directly - since we're building this from Python I'd just to that actually so that this here becomes way more readable.

it doesn't work as zbrush would see it as str unless you use strMerge.

Reply:
That seems odd. The `StrMerge` should also just be returning a string so the result should be the same.

Reply:
See the `ShellExecute` example here: https://docs.pixologic.com/user-guide/customizing-zbrush/zscripting/command-reference/

```
[ShellExecute, "open image.psd"]
```

Should work just fine.

Reply:
```
// Set a variable to " so we can quote the command line arguments for ShellExecute
[VarSet, q, [StrFromAsc, 34]]

[IPalette, "AYON", 1]

[ISubPalette, "AYON:Tools", 2]
[IButton, "AYON:Tools:Create...", "Open AYON Publisher on Create Tab",
	[ShellExecute, [StrMerge, #q, "{launch}", #q, "addon zbrush run-with-zscript --launcher create_tool"]], 0, 180
]

// Load
[IButton, "AYON:Tools:Load...", "Open AYON Loader",
	[ShellExecute, [StrMerge, #q, "{launch}", #q, "addon zbrush run-with-zscript --launcher loader_tool"]], 0, 180
]

// Publish
[IButton, "AYON:Tools:Publish...", "Open AYON Publisher",
	[ShellExecute, [StrMerge, #q, "{launch}", #q, "addon zbrush run-with-zscript --launcher publish_tool"]], 0, 180
]

// Manage
[IButton, "AYON:Tools:Manage...", "Open AYON Scene Inventory UI",
	[ShellExecute, [StrMerge, #q, "{launch}", #q, "addon zbrush run-with-zscript --launcher scene_inventory_tool"]], 0, 180
]

[ISubPalette, "AYON:Project", 2]
// Workfile
[IButton, "AYON:Project:Work Files...", "Open AYON Work Files UI",
	[ShellExecute, [StrMerge, #q, "{launch}", #q, "addon zbrush run-with-zscript --launcher workfiles_tool"]], 0, 180
]
```

![image](https://github.com/ynput/ayon-core/assets/2439881/5c12c927-ad82-4e6a-b588-3e5bf51d4b62)


This should work @moonyuet

Reply:
but the shellExecute code doesn't work, knowing that zscript only allows certain characters in each line

Reply:
the ui is successfully installed but the button cant trigger any tool